src/server/scripts/World/go_scripts.cpp

```diff
@@ -1569,6 +1570,131 @@ class go_pirate_day_music : public GameObjectScript
     }
 };
 
+/*####
+## go_bells
+####*/
+
+enum BellHourlySoundFX
+{
+    BELLTOLLHORDE          = 6595, // Horde
+    BELLTOLLTRIBAL         = 6675,
+    BELLTOLLALLIANCE       = 6594, // Alliance
+    BELLTOLLNIGHTELF       = 6674,
+    BELLTOLLDWARFGNOME     = 7234,
+    BELLTOLLKHARAZHAN      = 9154 // Kharazhan
+};
+
+enum BellHourlySoundAreas
+{
+    UNDERCITY_AREA         = 1497,
+    IRONFORGE_1_AREA       = 809,
+    IRONFORGE_2_AREA       = 1,
+    DARNASSUS_AREA         = 1657,
+    TELDRASSIL_ZONE        = 141,
+    KHARAZHAN_MAPID        = 532
+};
+
+enum BellHourlyObjects
+{
+    GO_HORDE_BELL          = 175885,
+    GO_ALLIANCE_BELL       = 176573,
+    GO_KHARAZHAN_BELL      = 182064
+};
+
+enum BellHourlyMisc
+{
+    GAME_EVENT_HOURLY_BELLS = 73
+};
+
+enum BellHourlyEvent
+{
+    EVENT_RING_BELL        = 1
+};
+
+class go_bells : public GameObjectScript
+{
+public:
+    go_bells() : GameObjectScript("go_bells") { }
+
+    struct go_bellsAI : public GameObjectAI
+    {
+        go_bellsAI(GameObject* go) : GameObjectAI(go), _rings(0)
+        {
+            InitializeAI();
+        }
+
+        void InitializeAI()
+        {
+            switch (go->GetEntry())
+            {
+                case GO_HORDE_BELL:
+                    _soundId = go->GetAreaId() == UNDERCITY_AREA ? BELLTOLLHORDE : BELLTOLLTRIBAL;
+                    break;
+                case GO_ALLIANCE_BELL:
+                {
+                    if (go->GetAreaId() == IRONFORGE_1_AREA || go->GetAreaId() == IRONFORGE_2_AREA)
+                        _soundId = BELLTOLLDWARFGNOME;
+                    else if (go->GetAreaId() == DARNASSUS_AREA || go->GetZoneId() == TELDRASSIL_ZONE)
+                        _soundId = BELLTOLLNIGHTELF;
+                    else
+                        _soundId = BELLTOLLALLIANCE;
+
+                    break;
+                }
+                case GO_KHARAZHAN_BELL:
+                {
+                    _soundId = BELLTOLLKHARAZHAN;
+                    break;
+                }
+            }
+        }
+
+        void OnGameEvent(bool start, uint16 eventId) override
+        {
+            if (eventId == GAME_EVENT_HOURLY_BELLS && start)
+            {
+                time_t time = sWorld->GetGameTime();
+                tm localTm;
+                localtime_r(&time, &localTm);
+                _rings = (localTm.tm_hour - 1) % 12 + 1;
+                _events.ScheduleEvent(EVENT_RING_BELL, Seconds(1));
+            }
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+            _events.Update(diff);
+
+            while (uint32 eventId = _events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                    case EVENT_RING_BELL:
+                    {
+                        go->PlayDirectSound(_soundId);
+
+                        for (auto i = 1; i < _rings + 1; ++i)
```

Comment:
wasn't it 3 seconds? also move to OnGameEvent :D
for (auto i = 0; i < _rings; ++i)
_events.ScheduleEvent(EVENT_RING_BELL, Seconds(i * 3 + 1));